### ISSUE TITLE
ValidatedEntry: Add "minimum_length" and "maximum_length" properties

### ISSUE THREAD

**Issue by jeremypw** [elementary/granite#489]
## Problem

In some cases it useful to set a minimum and/or maximum length properties for an input, in addition to other requirements.  Although this can be done with Regex and other ways, it would be nice to have an easier way.

## Proposal

Set these as properties on creation and leave it to the ValidatedEntry to handle their enforcement.

**Comment by danrabbit** [comment]
Gtk.Entry has `max_length` and `max_width_chars`. I think one of these properties should restrict character input already. But yeah a minimum length before the input is considered valid would be cool. We should probably base the name on whichever of these gtk properties does the thing we want

**Comment by cassidyjames** [comment]
You're looking for `max_length` which restricts the numbers of characters entered (whereas `max_width_chars` is the widget's maximum width in approximate characters, like when you set that on a label, I guess).

So `min_length` would be a great property to add.

**Comment by jeremypw** [comment]
Yes, I forgot that "max_length" was already a property of ValidatedEntry :grimacing:

**Comment by danrabbit** [comment]
Going to copy this relevant information from a recent fix in switchboard:

> I replaced the method `length` to check the number of characters with `char_count` because the original code assumed that every character was 1 byte, while in some languages some characters sizes could be more than 1 byte.
>
> For example this string is 8 bytes long, but only 4 characters long: áéóü.
